### PR TITLE
Run3-gex83C Convert one more SD (Hcal) to new format

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Convert one more SD (Hcal) to new format and move the one for CaloTrkProcessing to plugins

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
Please test

**Comment by bsunanda** [comment]
@smuzaffar @qliphy @silviodonato @perrotta The unit failure is not linked with this PR. It is in the current IB against which it is tested. Could care be taken for that?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch Hi Vladimir, you missed this one!

**Comment by civanch** [comment]
@bsunanda , what happens with this MR? Why regression is lost?

**Comment by civanch** [comment]
please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Calo/plugins/module.cc [hunk 1]
```diff
@@ -12,8 +11,6 @@
 
 typedef ECalSD EcalSensitiveDetector;
 DEFINE_SENSITIVEDETECTOR(EcalSensitiveDetector);
-typedef HCalSD HcalSensitiveDetector;
```
Comment (civanch) [inline SimG4CMS/Calo/plugins/module.cc:15]: @bsunanda , where HCalSD is now?

Comment (bsunanda) [reply SimG4CMS/Calo/plugins/module.cc:15]: HcalSensitiveDetectorBuilder instantiates HCalSD as HcalSensitiveDetector by supplying all objects needed from EventSetup

**Comment by bsunanda** [comment]
@civanch Please approve this. After it is integrated I can do this for other detectors in Calo

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**